packages/governance/src/tests/test_votes.cairo

```diff
@@ -0,0 +1,372 @@
+use openzeppelin_testing as utils;
+use openzeppelin_testing::constants::{SUPPLY, ZERO, DELEGATOR, DELEGATEE, OTHER};
+use openzeppelin_testing::events::EventSpyExt;
+use openzeppelin_token::erc20::ERC20Component::InternalTrait;
+use openzeppelin_token::erc721::ERC721Component::{
+    ERC721MetadataImpl, InternalImpl as ERC721InternalImpl,
+};
+use openzeppelin_token::erc721::ERC721Component::{ERC721Impl, ERC721CamelOnlyImpl};
+use openzeppelin_utils::cryptography::snip12::OffchainMessageHash;
+use openzeppelin_utils::structs::checkpoint::TraceTrait;
+use snforge_std::signature::stark_curve::{StarkCurveKeyPairImpl, StarkCurveSignerImpl};
+use snforge_std::{
+    start_cheat_block_timestamp_global, start_cheat_caller_address, spy_events, test_address
+};
+use snforge_std::{EventSpy};
+use starknet::ContractAddress;
+use starknet::storage::StoragePathEntry;
+
+const ERC_721_INITIAL_MINT: u256 = 10;
+
+//
+// Setup
+//
+
+type ComponentState = VotesComponent::ComponentState<ERC721VotesMock::ContractState>;
+type ERC20ComponentState = VotesComponent::ComponentState<ERC20VotesMock::ContractState>;
+
+fn COMPONENT_STATE() -> ComponentState {
+    VotesComponent::component_state_for_testing()
+}
+
+fn ERC20_COMPONENT_STATE() -> ERC20ComponentState {
+    VotesComponent::component_state_for_testing()
+}
+
+fn ERC721VOTES_CONTRACT_STATE() -> ERC721VotesMock::ContractState {
+    ERC721VotesMock::contract_state_for_testing()
+}
+
+fn ERC20VOTES_CONTRACT_STATE() -> ERC20VotesMock::ContractState {
+    ERC20VotesMock::contract_state_for_testing()
+}
+
+fn setup_erc721_votes() -> ComponentState {
+    let mut state = COMPONENT_STATE();
+    let mut mock_state = ERC721VOTES_CONTRACT_STATE();
+    // Mint ERC_721_INITIAL_MINT NFTs to DELEGATOR
+    let mut i: u256 = 0;
+    while i < ERC_721_INITIAL_MINT {
+        mock_state.erc721.mint(DELEGATOR(), i);
+        i += 1;
+    };
+    state
+}
+
+fn setup_erc20_votes() -> ERC20ComponentState {
+    let mut state = ERC20_COMPONENT_STATE();
+    let mut mock_state = ERC20VOTES_CONTRACT_STATE();
+
+    // Mint SUPPLY tokens to DELEGATOR
+    mock_state.erc20.mint(DELEGATOR(), SUPPLY);
+    state
+}
+
+fn setup_account(public_key: felt252) -> ContractAddress {
+    let mut calldata = array![public_key];
+    utils::declare_and_deploy("SnakeAccountMock", calldata)
+}
+
+//
+// Common tests for Votes
+//
+
+#[test]
+fn test_get_votes() {
+    let mut state = setup_erc721_votes();
+    start_cheat_caller_address(test_address(), DELEGATOR());
+    // Before delegating, the DELEGATOR has 0 votes
+    assert_eq!(state.get_votes(DELEGATOR()), 0);
+    state.delegate(DELEGATOR());
+
+    assert_eq!(state.get_votes(DELEGATOR()), ERC_721_INITIAL_MINT);
+}
+
+#[test]
+fn test_get_past_votes() {
+    let mut state = setup_erc721_votes();
+    let mut trace = state.Votes_delegate_checkpoints.entry(DELEGATOR());
+
+    start_cheat_block_timestamp_global('ts10');
+
```

Comment:
Went ahead and check the value at `ts0, which is after the mint but before the first manual checkpoints. The trace should still be zero, since we have not delegated.